dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -294,35 +291,56 @@ protected void addMetadata( EventQueryParams params, List<DimensionItemKeywords.
     {
         if ( !params.isSkipMeta() )
         {
-            final Map<String, Object> metadata = new HashMap<>();
+            Map<String, Object> metadata = new HashMap<>();
+            Map<String, List<Option>> dimensionOptions = getItemOptions( grid, params );
```

Comment:
Here we could use a ListValuedMap potentially.

https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/ListValuedMap.html